### PR TITLE
Ingrained - New Keyboard

### PR DESCRIPTION
**Pull request by jpuerto96**
## Description

This PR adds a new keyboard to QMK - Ingrained.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [ x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jpuerto-psc** [comment]
Bump for review 🙏🏽

File: keyboards/ingrained/ingrained.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define XXX KC_NO
+
+#define LAYOUT_3x6( \
```
Comment (drashna) [inline keyboards/ingrained/ingrained.h:18]:
```suggestion
#define LAYOUT_split_3x6_3( \
```

File: keyboards/ingrained/ingrained.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    { k00, k01, k02, k03, k04, k05 },\
+    { k10, k11, k12, k13, k14, k15 },\
+    { k20, k21, k22, k23, k24, k25 },\
+    { XXX, XXX, XXX, k33, k34, k35 },\
+    { k06, k07, k08, k09, k010, k011 },\
+    { k16, k17, k18, k19, k110, k111 },\
+    { k26, k27, k28, k29, k210, k211 },\
+    { XXX, XXX, XXX, k39, k310, k311 }\
+}
+#define LAYOUT_3x5( \
```
Comment (drashna) [inline keyboards/ingrained/ingrained.h:33]:
```suggestion
#define LAYOUT_split_3x5_3( \
```

File: keyboards/ingrained/keymaps/3x5/keymap.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names {
+  _BASE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_3x5(
```
Comment (drashna) [inline keyboards/ingrained/keymaps/3x5/keymap.c:17]:
```suggestion
  [_BASE] = LAYOUT_split_3x5_3(
```

File: keyboards/ingrained/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names {
+  _BASE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_3x6(
```
Comment (drashna) [inline keyboards/ingrained/keymaps/default/keymap.c:17]:
```suggestion
  [_BASE] = LAYOUT_split_3x6_3(
```

File: keyboards/ingrained/rules.mk [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = yes
+CUSTOM_MATRIX = lite
+NO_USB_STARTUP_CHECK = yes
+LTO_ENABLE = yes
+
+SRC += matrix.c
+QUANTUM_LIB_SRC += i2c_master.c
```
Comment (drashna) [inline keyboards/ingrained/rules.mk:26]:
```suggestion
QUANTUM_LIB_SRC += i2c_master.c

LAYOUTS = split_3x5_3 split_3x6_3
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/ingrained/rules.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/ingrained/rules.mk:16]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by jpuerto96** [comment]
@fauxpark ping for re-review.

**Comment by jpuerto96** [comment]
@fauxpark re-ping for re-review.

**Comment by jpuerto96** [comment]
@fauxpark pretty please review

**Comment by jpuerto96** [comment]
Ping for re reviewweww

**Comment by jpuerto96** [comment]
@fauxpark pretty please 😩

**Review by fauxpark — APPROVED** [review]

File: keyboards/ingrained/info.json [hunk 7]
```diff
@@ -0,0 +1,270 @@
+{
+    "keyboard_name": "ingrained",
+    "url": "https://github.com/jpuerto96",
+    "maintainer": "jpuerto96 (s8erdude)",
+    "layouts": {
+        "LAYOUT_3x6": {
```
Comment (drashna) [inline keyboards/ingrained/info.json:6]:
```suggestion
        "LAYOUT_split_3x6_3": {
```

File: keyboards/ingrained/info.json [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+                    "y": 3.7
+                },
+                {
+                    "label": "Alt",
+                    "x": 10,
+                    "y": 3.7
+                }
+            ]
+        },
+        "LAYOUT_3x5": {
```
Comment (drashna) [inline keyboards/ingrained/info.json:222]:
```suggestion
        "LAYOUT_split_3x5_3": {
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ingrained/readme.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+    make ingrained:default
+
+Flashing example for this keyboard:
+
+    make ingrained:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+To reset the board, you will want to short the pad labeled RESET on the main half (near the USB-C interconnect).
```
Comment (noroadsleft) [inline keyboards/ingrained/readme.md:19]:
```suggestion
To reset the board, you will want to short the pad labeled RESET on the main half (near the USB-C interconnect).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Just a minor change to the keyboard's readme - the links to the QMK Docs are always meant to be the last thing in this file.

**Review by noroadsleft — APPROVED** [review]

File: keyboards/ingrained/config.h [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xB33F
+#define PRODUCT_ID   0x58E4
+#define DEVICE_VER   0x0001
+#define MANUFACTURER s8erdude
+#define PRODUCT      ingrained
+
```
Comment (noroadsleft) [inline keyboards/ingrained/config.h:14]:
```suggestion
```

File: keyboards/ingrained/info.json [hunk 11]
```diff
@@ -0,0 +1,270 @@
+{
+    "keyboard_name": "ingrained",
+    "url": "https://github.com/jpuerto96",
+    "maintainer": "jpuerto96 (s8erdude)",
```
Comment (noroadsleft) [inline keyboards/ingrained/info.json:4]:
```suggestion
    "manufacturer": "s8erdude"
    "url": "https://github.com/jpuerto96",
    "maintainer": "jpuerto96 (s8erdude)",
    "usb": {
        "vid": "0xB33F",
        "pid": "0x58E4",
        "device_version": "0.0.1"
    },
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Spoke too soon.

This is a new guideline in QMK that will become required in 0.19.0 – USB Device Descriptors are now defined in `info.json` instead of `config.h`.

**Review by drashna — APPROVED** [review]

**Pull request merged**